src/Microsoft.TestPlatform.CrossPlatEngine/Discovery/DiscoveryManager.cs

```diff
@@ -247,6 +249,11 @@ private void TestSessionMessageHandler(object sender, TestRunMessageEventArgs e)
 
             if (this.testDiscoveryEventsHandler != null)
             {
+                if (e.Level == TestMessageLevel.Error)
```

Comment:
>Error [](start = 48, length = 5)

Do not downgrade error to warnings, this handler is also invoked when Adapters(Extensions) send something back as error, we cannot reduce their messages as warning.

Reply:
Removed Level downgrade lines, and now it follows whatever the Level it gets.